### PR TITLE
fix: remove internal alert stack and use the platform AlertStack

### PR DESCRIPTION
**Pull request by jenniferarnesen**
Fixes: https://dhis2.atlassian.net/browse/DHIS2-15438

The alerts were duplicated because there was both an internal alert stack/bar system as well as the built in one from app-runtime. Solution was to remove the internal alert stack/bar. This meant that the alerts coming from the layer loaders had to be refactored so that loader alerts are no longer are added to redux. Instead the component LayerLoaders use the app-runtime hook (useAlert).

Other changes:
* `skipPluginLogic: true` has been added to d2.config.js because of the cli-app-scripts upgrade. Previously we had set a resolution for the dependency to workaround this bug: https://dhis2.atlassian.net/browse/DHIS2-16916. Now, we can remove the resolution as long as we include this config.

- [ ] Testing: confirm that alert messages still show in the dashboard item legend.

Before (see video in Jira issue)

After: 
https://github.com/dhis2/maps-app/assets/6113918/309a29c4-e467-47a6-88e3-0e63ef5534b4

For reference, these are all the layer messages (where majority of changes are):

Earth Engine alerts:
image
image
image

Event Layer alerts:
image
image
image
image

Facility layer alerts
image
image

OrgUnit Layer alerts
image
image
image

Thematic Layer alerts
image
image
image
image

TE Layer alerts
image
image

### CODE REVIEW

File: src/loaders/facilityLoader.js [hunk 1]
```diff
@@ -80,11 +82,8 @@ const facilityLoader = async (config) => {
 
         if (!associatedGeometries.length) {
```
Comment (BRaimbault) [inline src/loaders/facilityLoader.js:83]: Similar to EarthEngineLoader.js L65, we get alerts for both OU missing geom and missing associated geom.
image

File: src/loaders/orgUnitLoader.js [hunk 2]
```diff
@@ -37,11 +42,8 @@ const orgUnitLoader = async (config) => {
             .catch((error) => {
                 if (error && error.message) {
```
Comment (BRaimbault) [inline src/loaders/orgUnitLoader.js:43]: API related errors do not seem to have a message property.

File: src/loaders/orgUnitLoader.js [hunk 3]
```diff
@@ -37,11 +42,8 @@ const orgUnitLoader = async (config) => {
...
-                        critical: true,
-                        message: i18n.t('Error: {{message}}', {
-                            message: error.message,
-                            nsSeparator: ';',
-                        }),
+                        code: ERROR_CRITICAL,
+                        message: error.message,
                     })
                 }
             }),
```
Comment (BRaimbault) [inline src/loaders/orgUnitLoader.js:49]: We should also catch errors from `getOrgUnitLevels(d2)` next line.

File: src/components/loaders/useLoaderAlerts.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        onHidden: loaderAlertAction,
+    })
+    const infoAlert = useAlert(ALERT_MESSAGE_DYNAMIC, {
+        info: true,
+        duration: 5000,
+        onHidden: loaderAlertAction,
+    })
+    const noDataAlert = useAlert(ALERT_MESSAGE_DYNAMIC, {
+        info: true,
+        duration: 5000,
```
Comment (BRaimbault) [inline src/components/loaders/useLoaderAlerts.js:28]: It seems a bit inconsistant to me that no geometry is getting the warning treatment and is persisting while no data is only an info and disappears.

File: d2.config.js [hunk 5]
```diff
@@ -20,6 +20,8 @@ const config = {
     },
 
     coreApp: true,
+
+    skipPluginLogic: true,
```
Comment (BRaimbault) [inline d2.config.js:24]: Why is this necessary?

File: src/loaders/trackedEntityLoader.js
Comment (BRaimbault) [inline src/loaders/trackedEntityLoader.js:1]: There are a number of other errors we could catch, but that's probably beyond the scope of this PR.

File: src/loaders/thematicLoader.js [hunk 6]
```diff
@@ -64,10 +70,8 @@ const thematicLoader = async (config) => {
                 ? {
                       alerts: [
                           {
-                              critical: true,
-                              message: `${i18n.t('Error')}: ${
-                                  error.message || error
-                              }`,
+                              code: ERROR_CRITICAL,
+                              message: error.message || error,
```
Comment (BRaimbault) [inline src/loaders/thematicLoader.js:74]: This works well, maybe it should be used in other layer loaders.

File: src/loaders/thematicLoader.js [hunk 7]
```diff
@@ -180,26 +184,21 @@ const thematicLoader = async (config) => {
...
             alerts.push({
-                warning: true,
-                message: `${name}: ${i18n.t('No data found')}`,
+                code: INFO_NO_DATA,
+                message: name,
             })
         }
     }
 
     if (coordinateField && !associatedGeometries.length) {
```
Comment (BRaimbault) [inline src/loaders/thematicLoader.js:198]: Similar to EarthEngineLoader.js L65, we get alerts for both OU missing geom and missing associated geom.

image

**Review by BRaimbault — CHANGES_REQUESTED** [review]
I belive some of the comments need to be addressed. Keep me posted.

File: d2.config.js (hunk 5 above)
Comment (jenniferarnesen) [reply d2.config.js:24]: Good catch, info added to the PR description.

File: src/components/loaders/useLoaderAlerts.js (hunk 4 above)
Comment (jenniferarnesen) [reply src/components/loaders/useLoaderAlerts.js:28]: I can definitely reconsider that. Currently "no data" is a warning, but that is not in line with our other apps, which simply present the empty chart/table/linelist, and "no data" not necessarily meaning there is something wrong. That's why I changed it. One improvement could be to add a "no data" indicator to the Layer Card? That could be implemented later once Joe is back.

You raise a good question as to whether "no geometry" and "no data" should behave the same. To me, they are slightly different. I was thinking of geometry as more of an instance configuration, the user was clearly expecting those to be defined and they weren't.

I could switch "no data" back to a warning, to be consistent with current behavior. Then we could address the way we do alerts altogether, once Joe is back.

File: src/loaders/earthEngineLoader.js [hunk 8]
```diff
@@ -62,22 +64,14 @@ const earthEngineLoader = async (config) => {
 
             if (!features.length) {
```
[the remark this answers is not in the document]
Comment (jenniferarnesen) [reply src/loaders/earthEngineLoader.js:65]: I was also not crazy about the double warning, however I was trying not to change the actual logic that generates the warnings.

File: src/loaders/trackedEntityLoader.js
Comment (jenniferarnesen) [reply src/loaders/trackedEntityLoader.js:1]: Yes. Let's address this when we can refactor the way we present errors to the user.

File: src/loaders/earthEngineLoader.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (BRaimbault) [reply src/loaders/earthEngineLoader.js:65]: Also for consistency, maybe the OU message could read: 
"Selected org units: No coordinates found"

[the remark this answers is not in the document]
Comment (jenniferarnesen) [reply src/loaders/earthEngineLoader.js:65]: Actually, one reason to leave this as is for now is because all the translations, would be made obsolete.

File: src/loaders/facilityLoader.js (hunk 1 above)
Comment (jenniferarnesen) [reply src/loaders/facilityLoader.js:83]: This also was on purpose - to avoid logic changes.

File: src/loaders/orgUnitLoader.js (hunk 3 above)
Comment (jenniferarnesen) [reply src/loaders/orgUnitLoader.js:49]: Let's do this after we have removed d2 (part of the remove schemas effort)

File: src/loaders/orgUnitLoader.js (hunk 2 above)
Comment (jenniferarnesen) [reply src/loaders/orgUnitLoader.js:43]: Fixed
